### PR TITLE
MPP-3183: Support JDK 11

### PR DESCRIPTION
**Pull request by yuokada**
## Working branches

- https://github.com/treasure-data/td-presto/tree/MPP-3183-jdk11a
- ~~https://github.com/treasure-data/td-presto/tree/MPP-3183-jdk11~~
- https://github.com/treasure-data/td-presto/tree/graviton2

----
I made a mistake.
https://github.com/prestosql/presto/pull/3791

### CODE REVIEW

**Comment by yuokada** [comment]
```
~/I/base-presto ❯❯❯ ./mvnw install -DskipTests
<!-- Cut a progress -->
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (default) @ presto-main ---
[INFO] Ignoring requireUpperBoundDeps in org.codehaus.plexus:plexus-utils
[WARNING] Invalid bytecodeVersion for module-info.class: expected 52, but was 53
[WARNING] Invalid bytecodeVersion for module-info.class: expected 52, but was 53
[WARNING] Invalid bytecodeVersion for module-info.class: expected 52, but was 53
[WARNING] Invalid bytecodeVersion for module-info.class: expected 52, but was 53
[WARNING] Invalid bytecodeVersion for module-info.class: expected 52, but was 53
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for javax.annotation:javax.annotation-api:1.3.1 paths to dependency are:
+-io.prestosql:presto-main:317
  +-javax.annotation:javax.annotation-api:1.3.1
and
+-io.prestosql:presto-main:317
  +-io.airlift:bootstrap:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:discovery:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:http-server:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2
and
+-io.prestosql:presto-main:317
  +-io.airlift.discovery:discovery-server:1.29
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:http-client:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:jaxrs:0.183.1
    +-org.glassfish.jersey.core:jersey-common:2.26
      +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:jaxrs:0.183.1
    +-org.glassfish.jersey.core:jersey-server:2.26
      +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:jaxrs:0.183.1
    +-org.glassfish.jersey.inject:jersey-hk2:2.26
      +-org.glassfish.hk2:hk2-locator:2.5.0-b42
        +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:jaxrs:0.183.1
    +-org.glassfish.jersey.inject:jersey-hk2:2.26
      +-org.glassfish.hk2:hk2-locator:2.5.0-b42
        +-org.glassfish.hk2:hk2-utils:2.5.0-b42
          +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.2
]
```

These 3 dependencies are problematic.
```
and
+-io.prestosql:presto-main:317
  +-io.airlift:discovery:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2
and
+-io.prestosql:presto-main:317
  +-io.airlift:http-server:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2

and
+-io.prestosql:presto-main:317
  +-io.airlift:http-client:0.183.1
    +-javax.annotation:javax.annotation-api:1.3.1 (managed) <-- javax.annotation:javax.annotation-api:1.3.2
```

**Comment by yuokada** [comment]
I downgraded javax.annotation-api to 1.3.1 and published 0.183.2.
https://github.com/treasure-data/airlift/commit/8fca1040b6817beceb68e45adc6a7e6b60c3cea7

**Comment by yuokada** [comment]
```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for io.airlift:log-manager:0.183 paths to dependency are:
+-io.prestosql:presto-phoenix:317
  +-io.airlift:bootstrap:0.183.2
    +-io.airlift:log-manager:0.183
and
+-io.prestosql:presto-phoenix:317
  +-io.prestosql:presto-main:317
    +-io.airlift:log-manager:0.183.2
and
+-io.prestosql:presto-phoenix:317
  +-io.prestosql:presto-tests:317
    +-io.airlift:log-manager:0.183.2
and
+-io.prestosql:presto-phoenix:317
  +-io.prestosql:presto-tests:317
    +-io.prestosql:presto-main:317
      +-io.airlift:log-manager:0.183.2
]
```

File: pom.xml [hunk 1]
```diff
@@ -44,8 +44,9 @@
         <air.maven.version>3.3.9</air.maven.version>
 
         <dep.antlr.version>4.7.1</dep.antlr.version>
-        <dep.airlift.version>0.183</dep.airlift.version>
-        <dep.packaging.version>${dep.airlift.version}</dep.packaging.version>
+        <dep.airlift.version>0.183.3</dep.airlift.version>
+        <!-- <dep.packaging.version>${dep.airlift.version}</dep.packaging.version> -->
+        <dep.packaging.version>0.183</dep.packaging.version>
```
Comment (xerial) [inline pom.xml]: Can't we use 0.183.3 here?

Comment (yuokada) [reply pom.xml]: I changed it to fix the following error.
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.4:single (package) on project presto-thrift-testing-server: Execution package of goal org.apache.maven.plugins:maven-assembly-plugin:2.4:single failed: Plugin org.apache.maven.plugins:maven-assembly-plugin:2.4 or one of its dependencies could not be resolved: Could not find artifact io.airlift:packaging:jar:0.183.3 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```
link: https://circleci.com/gh/treasure-data/presto/66?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Comment (yuokada) [reply pom.xml]: I needed to add `pluginRepositories` tag.

**Comment by yuokada** [comment]
```
May 28 07:54:42 ip-172-23-150-6 presto[30179]: + exec start-stop-daemon --start --chuid presto:presto --exec /opt/presto/bin/launcher.py -- run
May 28 07:54:42 ip-172-23-150-6 presto[30179]: OpenJDK 64-Bit Server VM warning: Option AggressiveOpts was deprecated in version 11.0 and will likely be removed in a future release.
May 28 07:54:42 ip-172-23-150-6 presto[30179]: Presto requires amd64 or ppc64le on Linux (found aarch64)
```
Need to backport this commit.
https://github.com/prestosql/presto/pull/2809

**Comment by yuokada** [comment]
Rebased from `hotfix-317` branch after a long time.

File: pom.xml [hunk 2]
```diff
@@ -44,7 +44,7 @@
         <air.maven.version>3.3.9</air.maven.version>
 
         <dep.antlr.version>4.7.1</dep.antlr.version>
-        <dep.airlift.version>0.183</dep.airlift.version>
+        <dep.airlift.version>0.183.5</dep.airlift.version>
```
Comment (yuokada) [inline pom.xml:47]: `0.183.5` is managed here.
https://github.com/treasure-data/airlift/commits/0.183.1-for-td

**Comment by yuokada** [comment]
I'm feeling like to merge this into `hotfix-317` branch around at when we completed plazma metadata api migration.

**Comment by yuokada** [comment]
This PR has been approved. I will merge.
https://treasure-data.slack.com/archives/G01242EDF2N/p1601965952072700?thread_ts=1601959087.070900&cid=G01242EDF2N

**Pull request merged**